### PR TITLE
Add spread to Operators page

### LINKED ISSUE #5599: Add missing spread operator on `Operators` page
### Page URL

https://dart.dev/language/operators/

### Page source

https://github.com/dart-lang/site-www/tree/main/./src/content/language/operators.md

### Describe the problem

This page doesn't mention spread operators ( https://dart.dev/language/collections#operators )

### Expected fix

_No response_

### Additional context

_No response_

### I would like to fix this problem.

- [ ] I will try and fix this problem on dart.dev.

### LINKED ISSUE #5635: [PAGE ISSUE]: 'Operators': The spread operators are missing
### Page URL

https://dart.dev/language/operators/

### Page source

https://github.com/dart-lang/site-www/tree/main/./src/content/language/operators.md

### Describe the problem

On page [Operators](https://dart.dev/language/operators) of the Tutorial, we are presented with a quite informative list of Dart operators.

However, it is still missing the new `...collection` and `...?collection` "spread operators" which inject collections into collections.

Additionally, the "if null" operator `a ?? b` is apparently generally called a ["null-coalescing"](https://en.wikipedia.org/wiki/Null_coalescing_operator) operator. One might want to note that.

### Expected fix

As described above

### Additional context

_No response_

### I would like to fix this problem.

- [ ] I will try and fix this problem on dart.dev.

### PR DESCRIPTION
**Pull request by MaryaBelanger**
I'd like to land #5639 first and update this PR accordingly. 

Fixes #5599 
Fixes #5635 

Adding spread to the table, and a section on the page. 

However, _the spread operator is not actually an operator expression_. So, it goes to the bottom of the precedence table and its section on the page basically just says "its part of collection syntax so read about them there" (taken from the [spec](https://github.com/dart-lang/language/blob/main/accepted/2.3/spread-collections/feature-specification.md#:~:text=It%20makes%20the,target%2C%20such%20as%3A))

### CODE REVIEW

**Comment by MaryaBelanger** [comment]
@munificent You wrote the spread spec, would you mind checking the language [here](https://dart-dev--pr5640-spread-operator-1wlk9h4h.web.app/language/operators#spread-operators)? (pretty much quoted directly from the spec)

File: src/content/language/operators.md [hunk 1]
```diff
@@ -19,24 +19,25 @@ and [operator precedence](#operator-precedence-example) from highest to lowest,
...
+| logical AND                             | `&&`                                                                                               | Left          |
+| logical OR                              | <code>&#124;&#124;</code>                                                                          | Left          |
+| if null                                 | `??`                                                                                               | Left          |
+| conditional                             | *`expr1`*    `?`    *`expr2`*    `:`    *`expr3`*                                                  | Right         |
+| cascade                                 | `..`    `?..`                                                                                      | Left          |
+| assignment                              | `=`    `*=`    `/=`    `+=`    `-=`    `&=`    `^=`    *etc.*                                      | Right         |
+| spread  ([See note](#spread-operators)) | `...`    `...?`                                                                                    | Left          |
```
Comment (munificent) [inline src/content/language/operators.md:40]: Spread is non-associative, so "Left" -> "None".

**Review by munificent — APPROVED** [review]
This looks great!

File: src/content/language/operators.md [hunk 2]
```diff
@@ -500,6 +501,25 @@ Strictly speaking, the "double dot" notation for cascades isn't an operator.
 It's just part of the Dart syntax.
 :::
 
+## Spread operators
+
+The spread operator evaluates an expression that yields a collection,
```
Comment (parlough) [inline src/content/language/operators.md:506]: nit: Switching from plural to singular felt a bit jarring to me. Consider sticking with one or the other.

**Review by parlough — APPROVED** [review]
Changes makes sense to me, mostly deferring to Bob :)

Thanks for updating this and clarifying its status/relation to other operator expressions.

File: src/content/language/operators.md (hunk 2 above)
Comment (MaryaBelanger) [reply src/content/language/operators.md:506]: Do you mean like, the heading is plural (`## Spread operators`) but the sentence is singular (`The spread operator evaluates...`)?

Comment (parlough) [reply src/content/language/operators.md:506]: Yes. As a reader unfamiliar with the spread operators it makes me wonder, what happened to the other spread operator(s)?

**Pull request merged**